react-components/toolbar-filter/step5.md

```diff
@@ -1,67 +1,120 @@
 
-## Task
-1) Locate the following "search-filter" `DataToolbarItem` and the "filter-group" `DataToolbarGroup`:
+1) **Define the required `onDelete` function to be passed to the DataToolbarFilters**
+
+Add it to the ToolbarDemoApp class below the `onSelect` function definition in the `ToolbarDemoApp` class.
+
+The `onDelete` callback function is executed whenever the user deletes a selected filter chip.
+
+<pre class="file" data-target="clipboard">
+onDelete = (type = &quot;&quot;, id = &quot;&quot;) =&gt; {
+  if (type) {
+    this.setState((prevState) =&gt; {
+      prevState.filters[type.toLowerCase()] = prevState.filters[type.toLowerCase()].filter(s =&gt; s !== id);
+      return {
+        filters: prevState.filters,
+      }
+    });
+  } else {
+    this.setState({
+      filters: {
+        name: [],
+        risk: [],
+        status: [],
+      }
+    })
+  }
+};
+</pre>
+
+2) **Locate the following three `DataToolbarItems` inside the `DataToolbarToggleGroup`:**
 
 <pre class="file">
-  &lt;DataToolbarItem variant=&quot;search-filter&quot;&gt;
-    &lt;Input value=&quot;&quot; id=&quot;step2Input&quot; ariaLabel=&quot;Step 2 input&quot; /&gt;
+&lt;DataToolbarItem variant=&quot;search-filter&quot;&gt;
+  &lt;Input value=&quot;&quot; id=&quot;step2Input&quot; ariaLabel=&quot;Step 2 input&quot; /&gt;
+&lt;/DataToolbarItem&gt;
+&lt;DataToolbarGroup variant=&quot;filter-group&quot;&gt;
+  &lt;DataToolbarItem&gt;
+    &lt;CheckboxSelect
+      onSelect={this.onSelect}
+      type=&quot;Status&quot;
+      selections={filters.status}
+      options={statusOptions}
+    /&gt;
   &lt;/DataToolbarItem&gt;
-  &lt;DataToolbarGroup variant=&quot;filter-group&quot;&gt;
-    &lt;DataToolbarItem&gt;
-      &lt;CheckboxSelect
-        onSelect={this.onSelect}
-        type=&quot;Status&quot;
-        selections={filters.status}
-        options={statusOptions}
-      /&gt;
-    &lt;/DataToolbarItem&gt;
-    &lt;DataToolbarItem&gt;
-      &lt;CheckboxSelect
-        onSelect={this.onSelect}
-        type=&quot;Risk&quot;
-        selections={filters.risk}
-        options={riskOptions}
-      /&gt;
-    &lt;/DataToolbarItem&gt;
-  &lt;/DataToolbarGroup&gt;
+  &lt;DataToolbarItem&gt;
+    &lt;CheckboxSelect
+      onSelect={this.onSelect}
+      type=&quot;Risk&quot;
+      selections={filters.risk}
+      options={riskOptions}
+    /&gt;
+  &lt;/DataToolbarItem&gt;
+&lt;/DataToolbarGroup&gt;
 </pre>
 
-2) Wrap both the `DataToolbarItem` and the `DataToolbarGroup` in the same `DataToolbarToggleGroup`. Provide both the `toggleIcon` and `breakpoint` properties to the `DataToolbarToggleGroup`.
+3) **Replace each of the three `DataToolbarItems` inside the `DataToolbarToggleGroup` with a `DataToolbarFilter`**
```

Comment:
This has potential to confuse the user, add an example of what it should look like at this step

Reply:
hm.... there will be no visible change...

Reply:
oh I see, like a code snippet. okay